### ISSUE TITLE
[Fatal] Call to a member function get_image_id() on bool | ProductGalleryUtils.php:85

### ISSUE THREAD

**Issue by nigeljamesstevenson** [woothemes/woocommerce#51154]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

We observed the following fatal error in relation to WooCommerce 9.2.3:

`Uncaught Error: Call to a member function get_image_id() on bool in /wordpress/plugins/woocommerce/9.2.3/src/Blocks/Utils/ProductGalleryUtils.php:85`

[Link to code](https://github.com/woocommerce/woocommerce/blob/6bf6f328546e35c143a5362a5bdeeafee5595f83/plugins/woocommerce/src/Blocks/Utils/ProductGalleryUtils.php#L85)

<details><summary>Stack trace</summary>

```
PHP Fatal error:  Uncaught Error: Call to a member function get_image_id() on bool in /.../plugins/woocommerce/9.2.3/src/Blocks/Utils/ProductGalleryUtils.php:85
Stack trace:
#0 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/ProductGallery.php(129): Automattic\WooCommerce\Blocks\Utils\ProductGalleryUtils::get_product_gallery_image_ids(false, 1)
#1 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/AbstractBlock.php(98): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render(Array, '\n\t<div class="w...', Object(WP_Block))
#2 /.../core/6.6.1/wp-includes/class-wp-block.php(519): Automattic\WooCommerce\Blocks\BlockTypes\AbstractBlock->render_callback(Array, '\n\t<div class="w...', Object(WP_Block))
#3 /.../core/6.6.1/wp-includes/class-wp-block.php(499): WP_Block->render()
#4 /.../core/6.6.1/wp-includes/blocks.php(2061): WP_Block->render()
#5 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/ProductGallery.php(63): render_block(Array)
#6 [internal function]: Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->Automattic\WooCommerce\Blocks\BlockTypes\{closure}('', Array)
#7 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/ProductGallery.php(65): array_reduce(Array, Object(Closure), '')
#8 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/ProductGallery.php(126): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render_dialog()
#9 /.../plugins/woocommerce/9.2.3/src/Blocks/BlockTypes/AbstractBlock.php(98): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render(Array, '\n<div class="wp...', Object(WP_Block))
#10 /.../core/6.6.1/wp-includes/class-wp-block.php(519): Automattic\WooCommerce\Blocks\BlockTypes\AbstractBlock->render_callback(Array, '\n<div class="wp...', Object(WP_Block))
#11 /.../core/6.6.1/wp-includes/blocks.php(2061): WP_Block->render()
#12 /.../core/6.6.1/wp-includes/blocks.php(2113): render_block(Array)
#13 /.../core/6.6.1/wp-includes/class-wp-hook.php(326): do_blocks('<!-- wp:woocomm...')
#14 /.../core/6.6.1/wp-includes/plugin.php(205): WP_Hook->apply_filters('<!-- wp:woocomm...', Array)
#15 /.../core/6.6.1/wp-includes/widgets/class-wp-widget-block.php(78): apply_filters('widget_block_co...', '<!-- wp:woocomm...', Array, Object(WP_Widget_Block))
#16 /.../core/6.6.1/wp-includes/class-wp-widget.php(394): WP_Widget_Block->widget(Array, Array)
#17 /.../core/6.6.1/wp-includes/widgets.php(845): WP_Widget->display_callback(Array, Array)
#18 /.../plugins/jetpack/13.8-a.7/jetpack_vendor/automattic/jetpack-search/src/class-helper.php(927): dynamic_sidebar('jetpack-instant...')
#19 /.../core/6.6.1/wp-includes/class-wp-hook.php(324): Automattic\Jetpack\Search\Helper::print_instant_search_sidebar('')
#20 /.../core/6.6.1/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#21 /.../core/6.6.1/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#22 /.../core/6.6.1/wp-includes/general-template.php(3081): do_action('wp_footer')
#23 /.../core/6.6.1/wp-includes/template-canvas.php(25): wp_footer()
#24 /.../core/6.6.1/wp-includes/template-loader.php(106): include('/.../core...')
#25 /.../core/6.6.1/wp-blog-header.php(19): require_once('/.../core...')
#26 /.../core/6.6.1/index.php(17): require('/.../core...')
#27 {main}
  thrown in /.../plugins/woocommerce/9.2.3/src/Blocks/Utils/ProductGalleryUtils.php on line 85
```

</details> 

### Expected behavior

No fatal errors when `$product->get_image_id();` is called.

### Actual behavior

Fatal errors when `$product->get_image_id();` is called.

### Steps to reproduce

We don't have steps to reproduce, but observed 1190 instances of this problem on Fri Aug 30, 1200 UTC to 2359 UTC on one site. Hopefully, someone can dig in and apply a defensive check if appropriate. - To be updated after investigation.

### WordPress Environment

N/A

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by Aljullu** [comment]
One way to reproduce is the following:

1. Go to Appearance > Editor > Patterns > Add New Pattern. Give a name to it and click on _Add_.
2. Open the inserter and search for _gallery_.
3. Add the Product Gallery (Beta) block to the pattern.

![imatge](https://github.com/user-attachments/assets/2ac637c1-b1ad-4e3e-879f-7f0d3c46e168)

4. Save it and notice the error.

**Comment by lanej0** [comment]
It looks like this issue may have crept back in again.

We observed the following fatal error in relation to WooCommerce 9.3.1:

Uncaught Error: Call to a member function get_image_id() on bool in /../../woocommerce/9.3.1/src/Blocks/Utils/ProductGalleryUtils.php:85

[Link to code](https://github.com/woocommerce/woocommerce/blob/6bf6f328546e35c143a5362a5bdeeafee5595f83/plugins/woocommerce/src/Blocks/Utils/ProductGalleryUtils.php#L85)

<details><summary>Stack trace</summary>
```
PHP Fatal error:  Uncaught Error: Call to a member function get_image_id() on bool in /../../woocommerce/9.3.1/src/Blocks/Utils/ProductGalleryUtils.php:85
Stack trace:
#0 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/ProductGallery.php(129): Automattic\WooCommerce\Blocks\Utils\ProductGalleryUtils::get_product_gallery_image_ids(false, 1)
#1 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/AbstractBlock.php(98): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render(Array, '\n\t<div class="w...', Object(WP_Block))
#2 /../core/6.6.2/wp-includes/class-wp-block.php(519): Automattic\WooCommerce\Blocks\BlockTypes\AbstractBlock->render_callback(Array, '\n\t<div class="w...', Object(WP_Block))
#3 /../core/6.6.2/wp-includes/class-wp-block.php(499): WP_Block->render()
#4 /../core/6.6.2/wp-includes/blocks.php(2061): WP_Block->render()
#5 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/ProductGallery.php(63): render_block(Array)
#6 [internal function]: Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->Automattic\WooCommerce\Blocks\BlockTypes\{closure}('', Array)
#7 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/ProductGallery.php(65): array_reduce(Array, Object(Closure), '')
#8 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/ProductGallery.php(126): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render_dialog()
#9 /../../woocommerce/9.3.1/src/Blocks/BlockTypes/AbstractBlock.php(98): Automattic\WooCommerce\Blocks\BlockTypes\ProductGallery->render(Array, '\n<div class="wp...', Object(WP_Block))
#10 /../core/6.6.2/wp-includes/class-wp-block.php(519): Automattic\WooCommerce\Blocks\BlockTypes\AbstractBlock->render_callback(Array, '\n<div class="wp...', Object(WP_Block))
#11 /../core/6.6.2/wp-includes/blocks.php(2061): WP_Block->render()
#12 /../core/6.6.2/wp-includes/blocks.php(2113): render_block(Array)
#13 /../core/6.6.2/wp-includes/class-wp-hook.php(326): do_blocks('<!-- wp:woocomm...')
#14 /../core/6.6.2/wp-includes/plugin.php(205): WP_Hook->apply_filters('<!-- wp:woocomm...', Array)
#15 /../core/6.6.2/wp-includes/widgets/class-wp-widget-block.php(78): apply_filters('widget_block_co...', '<!-- wp:woocomm...', Array, Object(WP_Widget_Block))
#16 /../core/6.6.2/wp-includes/class-wp-widget.php(394): WP_Widget_Block->widget(Array, Array)
#17 /../core/6.6.2/wp-includes/widgets.php(845): WP_Widget->display_callback(Array, Array)
#18 /../../jetpack/13.9-a.3/jetpack_vendor/automattic/jetpack-search/src/class-helper.php(927): dynamic_sidebar('jetpack-instant...')
#19 /../core/6.6.2/wp-includes/class-wp-hook.php(324): Automattic\Jetpack\Search\Helper::print_instant_search_sidebar('')
#20 /../core/6.6.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#21 /../core/6.6.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#22 /../core/6.6.2/wp-includes/general-template.php(3081): do_action('wp_footer')
#23 /../core/6.6.2/wp-includes/template-canvas.php(25): wp_footer()
#24 /../core/6.6.2/wp-includes/template-loader.php(106): include('/wordpress/core...')
#25 /../core/6.6.2/wp-blog-header.php(19): require_once('/wordpress/core...')
#26 /../core/6.6.2/index.php(17): require('/wordpress/core...')
#27 {main}
  thrown in /../../woocommerce/9.3.1/src/Blocks/Utils/ProductGalleryUtils.php on line 85
```
</summary>

Expected behavior
No fatal errors when $product->get_image_id(); is called.

Actual behavior
Fatal errors when $product->get_image_id(); is called.

Steps to reproduce
We don't have steps to reproduce, but observed 1091 instances of this problem on Fri Sep 13, 1200 UTC to 2359 UTC on one site. Hopefully, someone can dig in and apply a defensive check if appropriate. - To be updated after investigation.

**Comment by nigeljamesstevenson** [comment]
The [Fix fatal when adding the Product Gallery (Beta) block into a pattern](https://github.com/woocommerce/woocommerce/pull/51189#top) has a 9.4.0 Milestone so this should be resolved after that release next month 🤞

**Comment by Aljullu** [comment]
Closing because this was fixed in 9.4.0 as @nigeljamesstevenson said. Please reopen if you think it's a mistake.
